waku/v2/protocol/waku_enr.nim

```diff
@@ -66,9 +68,19 @@ func toCapabilities*(bitfield: CapabilitiesBitfield): seq[Capabilities] =
   toSeq(Capabilities.low..Capabilities.high).filterIt(supportsCapability(bitfield, it))
 
 
-## TODO: Turn into an EnrBuilder extension
-func toFieldPair*(caps: CapabilitiesBitfield): FieldPair =
-  toFieldPair(CapabilitiesEnrField, @[caps.uint8])
+# ENR builder extension
+
+proc withWakuCapabilities*(builder: var EnrBuilder, caps: CapabilitiesBitfield) =
+  builder.addFieldPair(CapabilitiesEnrField, @[caps.uint8])
+
+proc withWakuCapabilities*(builder: var EnrBuilder, caps: varargs[Capabilities]) =
+  withWakuCapabilities(builder, CapabilitiesBitfield.init(caps))
+
+proc withWakuCapabilities*(builder: var EnrBuilder, caps: openArray[Capabilities]) =
```

Comment:
just an opinion. I generally prefer one function rather than multiple ones. less code to read, more simple and clearer. not sure I see the value of having all three.

Isn't this too low level? Cool thing about the builder is abstracting away all this
`withWakuCapabilities(..., caps: CapabilitiesBitfield))`

Both are very similar.
`withWakuCapabilities(Relay, Store)`
`withWakuCapabilities(@[Relay, Store])`

tldr: I would just leave `withWakuCapabilities(@[Relay, Store])`

Reply:
Different `proc` signatures, cover different use cases. 

At first sight, you can see it as too low level, for example. But IIRC this is the exact API we need to create an ENR in the Waku discv5 module. A similar thing happens with the multiple capabilities methods: it is not the same to pass a `seq[T]` than a `varargs[T]`. You can can convert easily a `varargs` instance into a sequence, but the opposite has some compilation-time implications (varargs can be understood as an array of known size at compile time).